spec/models/audit/event/authn/login_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'spec_helper'
+
+describe Audit::Event::Authn::Login do
+  let(:role_id) { 'rspec:user:my_user' }
+  let(:role) { double('the-role', id: role_id) }
+  let(:authenticator_name) { 'the-authenticator'}
+  let(:service) { double('the-service', resource_id: 'rspec:webservice:the-service') }
```

Comment:
Another nitpick.  No need to change it here, but a small point of consistency worth doing in the future.  Note how we have `my_user` but `the-XXX` for the other objects.  I'd pick one or the other convention and use it everywhere.  My personal vote is for `my` because it immediately suggests "a variable thing" to me, but I don't care either way really.

Reply:
Separate point: Change in `authn.rb`: Pass `service_id` instead of `service`.  Same reasoning as `role` examples.